gui/unit-syndromes.lua

```diff
@@ -0,0 +1,543 @@
+    end,
+    [df.creature_interaction_effect_type.REMOVE_TAG] = function(effect)
+        local tags = {}
+
+        for tag, value in pairs(effect.tags1) do
+            if value then
+                table.insert(tags, string.upper(tag))
+            end
+        end
+
+        for tag, value in pairs(effect.tags2) do
+            if value then
+                table.insert(tags, string.upper(tag))
+            end
+        end
+
+        return ("REMOVES: \n%s"):format(table.concat(tags, "  \n"))
+    end,
+    [df.creature_interaction_effect_type.DISPLAY_TILE] = function(effect)
+        return ("TILE: %s %s"):format(effect.color, effect.tile)
+    end,
+    [df.creature_interaction_effect_type.FLASH_TILE] = function(effect)
+        return ("FLASH TILE: %s %s"):format(effect.sym_color[1], effect.sym_color[0])
+    end,
+    [df.creature_interaction_effect_type.SPEED_CHANGE] = function(effect)
+        local is_negative = effect.bonus_add < 0
+
+        if effect.bonus_add ~= 0 then
+            return ("SPEED: %s%s"):format(is_negative and "-" or "+", effect.bonus_add)
+        end
+
+        return ("SPEED: %s%%"):format(effect.bonus_perc)
+    end,
+    [df.creature_interaction_effect_type.CAN_DO_INTERACTION] = function(effect)
+        return ("ABILITY: %s"):format(getEffectInteraction(effect))
+    end,
+    [df.creature_interaction_effect_type.SKILL_ROLL_ADJUST] = function(effect)
+        return ("MODIFIER=%s, CHANGE=%s"):format(effect.multiplier, effect.chance)
+    end,
+    [df.creature_interaction_effect_type.BODY_TRANSFORMATION] = function(effect)
+        if effect.chance > 0 then
+            return ("%s, CHANCE=%s%%"):format(getEffectCreatureName(effect), effect.chance)
+        else
+            return getEffectCreatureName(effect)
+        end
+    end,
+    [df.creature_interaction_effect_type.PHYS_ATT_CHANGE] = function(effect)
+        return ("%s"):format(table.concat(getAttributePairs(effect.phys_att_add, effect.phys_att_perc), "\n"))
+    end,
+    [df.creature_interaction_effect_type.MENT_ATT_CHANGE] = function(effect)
+        return ("%s"):format(table.concat(getAttributePairs(effect.ment_att_add, effect.ment_att_perc), "\n"))
+    end,
+    -- TODO: Unfinished.
+    [df.creature_interaction_effect_type.MATERIAL_FORCE_MULTIPLIER] = function(effect)
+        local material = df.material[effect.mat_type]
+
+        return ("RECIEVED DAMAGE SCALED BY %s%%%s"):format(
+            (effect.fraction_mul * 100 / effect.fraction_div * 100) / 100,
+            material and ("vs. %s"):format(material.stone_name)
+        )
+    end,
+    -- TODO: Unfinished.
+    [df.creature_interaction_effect_type.BODY_MAT_INTERACTION] = function(effect)
+        return ("%s %s"):format(effect.interaction_name, effect.interaction_id)
+    end,
+    -- TODO: Unfinished.
+    [df.creature_interaction_effect_type.BODY_APPEARANCE_MODIFIER] = function(effect)
+        return ("TODO"):format(effect.interaction_name, effect.interaction_id)
```

Comment:
This line actually does look unfinished. are the other "Unfinished" comments still true?

Reply:
I checked all the other todos, there was one that was no longer true. The rest of the unfinished lines are either directly from the old script (like this one) or parts that I couldn't quite crack yet.